akka-remote/src/main/scala/akka/remote/artery/tcp/ssl/ConfigSSLEngineProvider.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2018-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.artery.tcp.ssl
+
+import java.security.SecureRandom
+
+import akka.actor.ActorSystem
+import akka.event.Logging
+import akka.event.MarkerLoggingAdapter
+import akka.remote.artery.tcp.SSLEngineProvider
+import akka.remote.artery.tcp.SecureRandomFactory
+import com.typesafe.config.Config
+import javax.net.ssl.SSLEngine
+import javax.net.ssl.SSLSession
+
+/**
+ * Config in akka.remote.artery.ssl.config-ssl-engine
+ */
+final class ConfigSSLEngineProvider(protected val config: Config, protected val log: MarkerLoggingAdapter)
+    extends SSLEngineProvider {
+
+  def this(system: ActorSystem) =
+    this(
+      system.settings.config.getConfig("akka.remote.artery.ssl.config-ssl-engine"),
+      Logging.withMarker(system, classOf[ConfigSSLEngineProvider].getName))
+
+  private val rng: SecureRandom = SecureRandomFactory.createSecureRandom(config, log)
+  // Creating a new factory requires creating a new managersProvider so certificates are freshly read.
+  def providerFactory: (Config) => SslManagersProvider = new JksManagersProvider(_)
+  val sessionVerifierFactory: SslManagersProvider => SessionVerifier = _ => NoopSessionVerifier
```

Comment:
Those functions for `providerFactory` and sessionVerifierFactory might be difficult for users to understand how to override if `ConfigSSLEngineProvider` is intended to be subclassed. Also doesn't work for Java. Could we make them methods?